File: A6/src/repository/IRepository.java
package repository;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import exp.PrgState;
import model.*;
public interface IRepository
{
	void add(PrgState prg);
	ArrayList<PrgState> getPrgList();
	void setPrgList(List<PrgState> prgList);
	//PrgState getCrtPrg();
	void logPrgStateExec(PrgState prg) throws IOException;
}

File: A7/Rep/IRepository.cs
using A7.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace A7.Rep
{
    interface IRepository
    {
        void Add(PrgState prg);
        PrgState GetCrtPrg();
        void LogPrgStateExec();
    }
}
